File: Hotel/csharp/HOTELINTERFAZ/HOTELINTERFAZ/Models/Habitacion.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace HOTELINTERFAZ.Models
{
    public class Habitacion
    {
        [JsonPropertyName("_id")]
        public string Id { get; set; }

        [JsonPropertyName("numero")]
        public int Numero { get; set; }

        [JsonPropertyName("tipo")]
        public string Tipo { get; set; } = "";

        [JsonPropertyName("descripcion")]
        public string Descripcion { get; set; } = "";

        [JsonPropertyName("imagen")]
        public string Imagen { get; set; } = "";

        [JsonPropertyName("precionoche")]
        public double PrecioNoche { get; set; }

        [JsonPropertyName("rate")]
        public double Rate { get; set; }

        [JsonPropertyName("max_ocupantes")]
        public int MaxOcupantes { get; set; }

        [JsonPropertyName("disponible")]
        public bool Disponible { get; set; }

        [JsonPropertyName("oferta")]
        public bool Oferta { get; set; }

        [JsonPropertyName("servicios")]
        public ObservableCollection<string> Servicios { get; set; } = new();

        [JsonPropertyName("amenities")]
        public ObservableCollection<Amenity> Amenities { get; set; } = new();

        [JsonPropertyName("pets_allowed")]
        public bool PetsAllowed { get; set; }

        [JsonPropertyName("pet_supplement_per_night")]
        public double PetSupplementPerNight { get; set; }

        [JsonPropertyName("aceptaMascotas")]
        public bool AceptaMascotas { get; set; }

        [JsonPropertyName("suplementoMascota")]
        public double SuplementoMascota { get; set; }

        [JsonPropertyName("politicaMascotas")]
        public string PoliticaMascotas { get; set; } = "";

        [JsonPropertyName("maxMascotas")]
        public int MaxMascotas { get; set; }

        [JsonIgnore]
        public bool AdmiteMascotas
        {
            get => PetsAllowed || AceptaMascotas;
            set
            {
                PetsAllowed = value;
                AceptaMascotas = value;
            }
        }

        [JsonIgnore]
        public double SuplementoMascotasNoche
        {
            get => PetSupplementPerNight > 0 ? PetSupplementPerNight : SuplementoMascota;
            set
            {
                PetSupplementPerNight = value;
                SuplementoMascota = value;
            }
        }

        [JsonIgnore]
        public string AmenitiesTexto => Amenities == null || Amenities.Count == 0
            ? ""
            : string.Join(", ", Amenities.Select(a => a.Name));

        [JsonIgnore]
        public List<string> AmenityIds => Amenities?
            .Where(a => !string.IsNullOrWhiteSpace(a.Id))
            .Select(a => a.Id)
            .ToList() ?? new List<string>();
        
        [JsonIgnore]
        public string ServiciosTexto
        {
            get => Servicios == null || Servicios.Count == 0
                ? ""
                : string.Join(", ", Servicios);
            set
            {
                Servicios ??= new ObservableCollection<string>();
                Servicios.Clear();

                if (string.IsNullOrWhiteSpace(value))
                    return;

                var tokens = value
                    .Split(new[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries)
                    .Select(s => s.Trim())
                    .Where(s => !string.IsNullOrWhiteSpace(s));

                foreach (var t in tokens)
                    Servicios.Add(t);
            }
        }

    }
}

File: Hotel/kotlin/app/src/main/java/com/example/aplicacion_hotel/Model/Habitacion.kt
package com.example.aplicacion_hotel.Model

data class Amenity(
    val _id: String = "",
    val name: String = "",
    val icon: String = "",
    val category: String = ""
)

data class PetPolicy(
    val habitacionId: String = "",
    val numero: Int = 0,
    val tipo: String = "",
    val pets_allowed: Boolean = false,
    val pet_supplement_per_night: Double = 0.0,
    val aceptaMascotas: Boolean = false,
    val politicaMascotas: String = "",
    val suplementoMascota: Double = 0.0,
    val maxMascotas: Int = 0
) {
    val admiteMascotas: Boolean
        get() = pets_allowed || aceptaMascotas

    val suplementoMascotaNoche: Double
        get() = listOf(pet_supplement_per_night, suplementoMascota).firstOrNull { it > 0 } ?: 0.0
}

data class Habitacion(
    val _id: String,
    val numero: Int,
    val tipo: String,
    val descripcion: String,
    val imagen: String = "",
    val imagenes: List<String> = emptyList(),
    val precionoche: Double,
    val rate: Double,
    val max_ocupantes: Int,
    val disponible: Boolean,
    val oferta: Boolean,
    val amenities: List<Amenity> = emptyList(),
    val pets_allowed: Boolean = false,
    val pet_supplement_per_night: Double = 0.0,
    val aceptaMascotas: Boolean = false,
    val politicaMascotas: String = "",
    val suplementoMascota: Double = 0.0,
    val maxMascotas: Int = 0,
    val servicios: List<String> = emptyList()
) {
    val admiteMascotas: Boolean
        get() = pets_allowed || aceptaMascotas

    val suplementoMascotaNoche: Double
        get() = listOf(pet_supplement_per_night, suplementoMascota).firstOrNull { it > 0 } ?: 0.0
}
